tests/integration-tests/framework/credential_providers.py

```diff
@@ -25,15 +25,19 @@ def register_cli_credentials_for_region(region, iam_role):
 
 def run_pcluster_command(*args, **kwargs):
     """Run a command after assuming the role configured through register_cli_credentials_for_region."""
+
     region = kwargs.get("region")
     if not region:
         region = os.environ["AWS_DEFAULT_REGION"]
 
-    if region in cli_credentials:
-        with sts_credential_provider(region, cli_credentials[region]):
+        if region in cli_credentials:
+            with sts_credential_provider(
+                region, credential_arn=kwargs.get("custom_cli_credentials") or cli_credentials.get(region)
+            ):
+                kwargs.pop("custom_cli_credentials", None)
+                return run_command(*args, **kwargs)
+        else:
```

Comment:
did you mean for this to be indented under the `if not region`?

Reply:
Reverted the changes